via/services/youtube.py

```diff
@@ -0,0 +1,23 @@
+from urllib.parse import parse_qs, urlparse
+
+
+class YoutubeService:
+    @classmethod
+    def parse_url(cls, public_url):
+        """Get the youtube video ID from an URL."""
+        parsed = urlparse(public_url)
```

Comment:
This is a bit wordy compared to just a regexp but easier to get right and to read IMO.